client/components/hoc/protect-form/README.md

```diff
@@ -0,0 +1,34 @@
+Protect-Form
+============
+
+This HoC protects a user from navigating away from a form that has been edited without having been saved. When your form has been marked as changed, the browser will open a confirm dialog if you try to navigate away from the page without saving the changes.
```

Comment:
We should be more verbose with terminology. I expect many developers won't understand `HoC`. At the very least we should expand it out to its full form, maybe also link to a good resource on explaining higher-order components.